docker/windows/Scripts/configure-tentacle.ps1

```diff
@@ -146,6 +157,13 @@ function Validate-Variables() {
     }
   }
 
+  if($TargetTenant -ne $null -or $TargetTenantTag -ne $null) {
+    if($TargetTenantedDeploymentParticipation -ne "Tenanted" -or $TargetTenantedDeploymentParticipation -ne "TenantedOrUntenanted"){
+      Write-Error "The 'TargetTenant' and 'TargetTenantTag' environment variables are not valid when the 'TargetTenantedDeploymentParticipation' variable is not set to 'Tenanted' or 'TenantedOrUntenanted'"
```

Comment:
Same question as above re requiring participation to be set explicitly.

Reply:
Thinking about it a bit more, I wonder if the validation is inside out. If tenanted deployment participation is specified, ensure a tenant or tag has also been specified.

Reply:
.. and some more. I wonder why we are performing this type of validation at all, instead of letting the register command do it?

Reply:
I didn't check the behavior of the register-with command - I saw that the script already had some validation and thought of including some for the tenant variables. But if the register command handles all this anyway, then I agree it doesn't seem like any validation would be needed in this script.

Reply:
I think there's some validation in this script since the messages that are returned from the commands themselves don't match with what environment variables the user needs to provide via the docker command. Eg. if you don't provide `ServerUrl` (the docker environment variable name) with no validation the command returns this message:
```
Please specify an Octopus Server, e.g., --server=http://your-octopus-server
```
Whereas it currently says:
```
Missing 'ServerUrl' environment variable
```
The second one seems seems a bit clearer, otherwise the user would need to check what `server` is on the Docker image. Though I like the first error message more just in terms of how it reads.

There's another example which definitely should have some sort of validation, where if you don't specify a listening/server port you get:
```
Unrecognized command line arguments: False
```
which is a bit too unclear.

So I think certain validation should stay to help the user understand what they're missing, but checking the logic like whether tenant/tag exist with deployment participation can probably be left to the command itself.

Reply:
Actually it looks like the Windows docker image has a default listening port but the Linux one doesn't. So that case shouldn't require any validation, it's just that the Linux image is missing a default value.